tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -302,73 +278,53 @@ void CustomActionData::findSuppliedUserInfo(std::wstring &input, std::wstring &c
 
 bool CustomActionData::parseUsernameData()
 {
-    std::wstring tmpName = ddAgentUserName;
-    bool previousInstall = findPreviousUserInfo();
-    bool userSupplied = false;
-
-    if (this->value(propertyDDAgentUserName, tmpName))
-    {
-        if (tmpName.length() == 0)
-        {
-            tmpName = ddAgentUserName;
-        }
-        else
-        {
-            userSupplied = true;
-        }
-    }
-    if (std::wstring::npos == tmpName.find(L'\\'))
-    {
-        WcaLog(LOGMSG_STANDARD, "loaded username doesn't have domain specifier, assuming local");
-        tmpName = L".\\" + tmpName;
-    }
-    // now create the splits between the domain and user for all to use, too
-    std::wstring computed_domain, computed_user;
+    auto userFromPreviousInstall= findPreviousUserInfo();
+    auto userFromCommandLine = findSuppliedUserInfo();
 
     // if this is an upgrade (we found a previously recorded username in the registry)
     // and nothing was supplied on the command line, don't bother computing that.  Just use
     // the existing
-    if (previousInstall && !userSupplied)
+    if (userFromPreviousInstall && !userFromCommandLine)
     {
-        computed_domain = pvsDomain;
-        computed_user = pvsUser;
         WcaLog(LOGMSG_STANDARD, "Using username from previous install");
+        _user = userFromPreviousInstall.value();
+    }
+    else if (userFromCommandLine)
+    {
+        _user = userFromCommandLine.value();
     }
     else
     {
-        findSuppliedUserInfo(tmpName, computed_domain, computed_user);
-        checkForUserMismatch(previousInstall, userSupplied, computed_domain, computed_user);
+        // Didn't find a user in the registry nor from the command line
+        // use default value. Order of construction is Domain then Name
+        _user = {L".", ddAgentUserName };
     }
 
-    _domain = computed_domain;
-    _fqUsername = computed_domain + L"\\" + computed_user;
-    _unqualifiedUsername = computed_user;
+    ensureDomainHasCorrectFormat();
```

Comment:
This function name makes it look like a validation to me, but it actually changes the domain. Maybe it could be called `resolveDomainName()` or something like that?